{
	"document_title": "CVE-2019-19588",
	"document_type": "SUSE CVE",
	"document_publisher": {
		"type": "Vendor",
		"contact_details": "security@suse.de",
		"issuing_authority": "SUSE Security Team"
	},
	"document_tracking": {
		"identification": {
			"id": "SUSE CVE-2019-19588"
		},
		"status": "Interim",
		"version": "1",
		"revision_history": {
			"revision": {
				"number": "7",
				"date": "2024-10-31T01:48:04Z",
				"description": "current"
			}
		},
		"initial_release_date": "2021-05-30T14:33:45Z",
		"current_release_date": "2024-10-31T01:48:04Z",
		"generator": {
			"engine": "cve-database/bin/generate-cvrf-cve.pl",
			"date": "2020-12-27T01:00:00Z"
		}
	},
	"document_notes": [
		{
			"text": "CVE-2019-19588",
			"title": "CVE",
			"type": "Summary"
		},
		{
			"text": "The validators package 0.12.2 through 0.12.5 for Python enters an infinite loop when validators.domain is called with a crafted domain string. This is fixed in 0.12.6.",
			"title": "Mitre CVE Description",
			"type": "Description"
		},
		{
			"text": "The CVRF data is provided by SUSE under the Creative Commons License 4.0 with Attribution (CC-BY-4.0).",
			"title": "Terms of Use",
			"type": "Legal Disclaimer"
		}
	],
	"document_references": [
		{
			"type": "Self",
			"url": "https://www.suse.com/support/security/rating/",
			"description": "SUSE Security Ratings"
		}
	],
	"product_tree": {
		"branch": [
			{
				"type": "Product Family",
				"name": "SUSE Linux Enterprise Desktop 15 SP6",
				"branch": [
					{
						"type": "Product Name",
						"name": "SUSE Linux Enterprise Module for Python 3 15 SP6",
						"full_product_name": {
							"text": "SUSE Linux Enterprise Module for Python 3 15 SP6",
							"product_id": "SUSE Linux Enterprise Module for Python 3 15 SP6",
							"cpe": "cpe:/o:suse:sle-module-python3:15:sp6"
						}
					}
				],
				"full_product_name": {}
			},
			{
				"type": "Product Family",
				"name": "SUSE Linux Enterprise High Performance Computing 15 SP6",
				"branch": [
					{
						"type": "Product Name",
						"name": "SUSE Linux Enterprise Module for Python 3 15 SP6",
						"full_product_name": {
							"text": "SUSE Linux Enterprise Module for Python 3 15 SP6",
							"product_id": "SUSE Linux Enterprise Module for Python 3 15 SP6",
							"cpe": "cpe:/o:suse:sle-module-python3:15:sp6"
						}
					}
				],
				"full_product_name": {}
			},
			{
				"type": "Product Family",
				"name": "SUSE Linux Enterprise Server 15 SP6",
				"branch": [
					{
						"type": "Product Name",
						"name": "SUSE Linux Enterprise Module for Python 3 15 SP6",
						"full_product_name": {
							"text": "SUSE Linux Enterprise Module for Python 3 15 SP6",
							"product_id": "SUSE Linux Enterprise Module for Python 3 15 SP6",
							"cpe": "cpe:/o:suse:sle-module-python3:15:sp6"
						}
					}
				],
				"full_product_name": {}
			},
			{
				"type": "Product Family",
				"name": "SUSE Linux Enterprise Server for SAP Applications 15 SP6",
				"branch": [
					{
						"type": "Product Name",
						"name": "SUSE Linux Enterprise Module for Python 3 15 SP6",
						"full_product_name": {
							"text": "SUSE Linux Enterprise Module for Python 3 15 SP6",
							"product_id": "SUSE Linux Enterprise Module for Python 3 15 SP6",
							"cpe": "cpe:/o:suse:sle-module-python3:15:sp6"
						}
					}
				],
				"full_product_name": {}
			},
			{
				"type": "Product Family",
				"name": "openSUSE Tumbleweed",
				"branch": [
					{
						"type": "Product Name",
						"name": "openSUSE Tumbleweed",
						"full_product_name": {
							"text": "openSUSE Tumbleweed",
							"product_id": "openSUSE Tumbleweed",
							"cpe": "cpe:/o:opensuse:tumbleweed"
						}
					}
				],
				"full_product_name": {}
			},
			{
				"type": "Product Version",
				"name": "python310-validators-0.24.0-1.2",
				"full_product_name": {
					"text": "python310-validators-0.24.0-1.2",
					"product_id": "python310-validators-0.24.0-1.2"
				}
			},
			{
				"type": "Product Version",
				"name": "python311-validators-0.20.0-150400.9.3.9",
				"full_product_name": {
					"text": "python311-validators-0.20.0-150400.9.3.9",
					"product_id": "python311-validators-0.20.0-150400.9.3.9"
				}
			},
			{
				"type": "Product Version",
				"name": "python311-validators-0.24.0-1.2",
				"full_product_name": {
					"text": "python311-validators-0.24.0-1.2",
					"product_id": "python311-validators-0.24.0-1.2"
				}
			},
			{
				"type": "Product Version",
				"name": "python312-validators-0.24.0-1.2",
				"full_product_name": {
					"text": "python312-validators-0.24.0-1.2",
					"product_id": "python312-validators-0.24.0-1.2"
				}
			},
			{
				"type": "Product Version",
				"name": "python36-validators-0.18.1-2.4",
				"full_product_name": {
					"text": "python36-validators-0.18.1-2.4",
					"product_id": "python36-validators-0.18.1-2.4"
				}
			},
			{
				"type": "Product Version",
				"name": "python38-validators-0.18.1-2.4",
				"full_product_name": {
					"text": "python38-validators-0.18.1-2.4",
					"product_id": "python38-validators-0.18.1-2.4"
				}
			},
			{
				"type": "Product Version",
				"name": "python39-validators-0.18.1-2.4",
				"full_product_name": {
					"text": "python39-validators-0.18.1-2.4",
					"product_id": "python39-validators-0.18.1-2.4"
				}
			}
		],
		"relationship": [
			{
				"product_reference": "python311-validators-0.20.0-150400.9.3.9",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Module for Python 3 15 SP6"
			},
			{
				"product_reference": "python310-validators-0.24.0-1.2",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "openSUSE Tumbleweed"
			},
			{
				"product_reference": "python311-validators-0.24.0-1.2",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "openSUSE Tumbleweed"
			},
			{
				"product_reference": "python312-validators-0.24.0-1.2",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "openSUSE Tumbleweed"
			},
			{
				"product_reference": "python36-validators-0.18.1-2.4",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "openSUSE Tumbleweed"
			},
			{
				"product_reference": "python38-validators-0.18.1-2.4",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "openSUSE Tumbleweed"
			},
			{
				"product_reference": "python39-validators-0.18.1-2.4",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "openSUSE Tumbleweed"
			}
		]
	},
	"vulnerability": [
		{
			"notes": [
				{
					"text": "The validators package 0.12.2 through 0.12.5 for Python enters an infinite loop when validators.domain is called with a crafted domain string. This is fixed in 0.12.6.",
					"title": "Vulnerability Description",
					"type": "General"
				}
			],
			"cve": "CVE-2019-19588",
			"product_statuses": [
				{
					"type": "Fixed",
					"product_id": [
						"SUSE Linux Enterprise Module for Python 3 15 SP6:python311-validators-0.20.0-150400.9.3.9",
						"openSUSE Tumbleweed:python310-validators-0.24.0-1.2",
						"openSUSE Tumbleweed:python311-validators-0.24.0-1.2",
						"openSUSE Tumbleweed:python312-validators-0.24.0-1.2",
						"openSUSE Tumbleweed:python36-validators-0.18.1-2.4",
						"openSUSE Tumbleweed:python38-validators-0.18.1-2.4",
						"openSUSE Tumbleweed:python39-validators-0.18.1-2.4"
					]
				}
			],
			"threats": [
				{
					"type": "Impact",
					"description": "important"
				}
			],
			"cvss_score_sets": {
				"score_set_v_2": {
					"base_score_v_2": "7.8",
					"vector_v_2": "AV:N/AC:L/Au:N/C:N/I:N/A:C"
				},
				"score_set_v_3": {
					"base_score_v_3": "7.5",
					"vector_v_3": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H"
				}
			}
		}
	]
}
